recommendation/search/recommendation.py

```diff
@@ -16,8 +16,12 @@ class SearchRecommendation(object):
       sequentially determine if they match the result, then enhance the result
       if so.
     """
-    def __init__(self, query):
+    def __init__(self, query, debug=False):
+        self.debug = debug
         self.query = query
+        self.recommendation = None
+        self.suggestion_engine = self.get_suggestion_engine()
+        self.query_engine = self.get_query_engine()
```

Comment:
Sure, I thought as much, but isn't that what `@property` is for ;)

This just struck me as odd, as you later on changed the code to use `self.suggestion_engine` directly, so it isn't either getter-style nor proprerty-style but both at the same time.